apiserver/facades/agent/uniter/uniter.go

```diff
@@ -168,19 +180,23 @@ func (u *UniterAPI) OpenedMachinePortRangesByEndpoint(ctx context.Context, args
 	return result, nil
 }
 
-func (u *UniterAPI) getOneMachineOpenedPortRanges(canAccess common.AuthFunc, machineTag string) (state.MachinePortRanges, error) {
+func (u *UniterAPI) getOneMachineOpenedPortRanges(ctx context.Context, canAccess common.AuthFunc, machineTag string) (map[coreunit.UUID]network.GroupedPortRanges, error) {
 	tag, err := names.ParseMachineTag(machineTag)
 	if err != nil {
 		return nil, apiservererrors.ErrPerm
 	}
 	if !canAccess(tag) {
 		return nil, apiservererrors.ErrPerm
 	}
-	machine, err := u.getMachine(tag)
+	machineUUID, err := u.machineService.GetMachineUUID(ctx, coremachine.Name(tag.Id()))
 	if err != nil {
-		return nil, err
+		return nil, internalerrors.Errorf("cannot get machine UUID for %q: %w", tag, err)
```

Comment:
Again, "getting machine UUID...". The fact that you can not, is implied by the error condition.

Reply:
Done